lab-katherine/src/main.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import superagent from 'superagent'
+
+class SearchResultList extends React.Component {
+  render(){
+    let topics = this.props.topics
+    return (
+      <ul>
+        {topics.map((result, i) =>
+          <li key={i}>
+            <p> {result.title} </p>
+            <p> up votes: {result.ups} </p>
+            <a href={result.url}> click me </a>
+          </li>
+        )}
+      </ul>
+    )
+  }
+}
+
+class SearchForm extends React.Component {
+  constructor(props){
+    super(props)
+    this.state = {
+      searchFormBoard: '',
+      searchFormLimit: 0,
+    }
+
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.handleChange = this.handleChange.bind(this)
+  }
+
+  handleSubmit(e){
+    e.preventDefault()
+    this.props.onComplete(this.state)
```

Comment:
This line is not sending the right query in my tests. I get : https://www.reddit.com/r/[object%20Object].json?limit=undefined. Please check how you send the parameters to the onComplete function. They need to be either destructured or sent independently.